src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1743,12 +1744,16 @@ private object TransformValue(WhereParams whereParams, bool forRange = false)
 
         private string AddQueryParameter(object value)
         {
-            var parameterName = $"{ParameterPrefix}{QueryParameters.Count.ToInvariantString()}";
+            string parameterName;
+            if (ParameterPrefix == "p" && QueryParameters.Count < 128)
+                parameterName = _pregeneratedParametersForPrefixP[QueryParameters.Count];
+            else
+                parameterName = $"{ParameterPrefix}{QueryParameters.Count.ToInvariantString()}";
             QueryParameters.Add(parameterName, value);
             return parameterName;
         }
 
-        public string ParameterPrefix { get; set; } = "p";
+        public string ParameterPrefix { get; set; }  = "p";
```

Comment:
introduce prvate const inside this class `DefaultParameterPrefix` and set that to `p` and use it both places.

Reply:
done